sktime/clustering/spatio_temporal/_st_dbscan.py

```diff
@@ -0,0 +1,296 @@
+"""
+ST-DBSCAN - fast scalable implementation of ST DBSCAN
+            scales also to memory by splitting into frames
+            and merging the clusters together
+"""
+
+__author__ = ["VectorNd"]
+
+from sktime.clustering import BaseClusterer
+
+import numpy as np
+
+from scipy.spatial.distance import pdist, squareform
+from scipy.sparse import coo_matrix
+
+from sklearn.cluster import DBSCAN
+from sklearn.utils import check_array
+from sklearn.neighbors import NearestNeighbors
+
+import warnings
+
+class ST_DBSCAN(BaseClusterer):
+    """
+    A class to perform the ST_DBSCAN clustering
+    Parameters
+    ----------
+    eps1 : float, default=0.5
+        The spatial density threshold (maximum spatial distance) between 
+        two points to be considered related.
+    eps2 : float, default=10
+        The temporal threshold (maximum temporal distance) between two 
+        points to be considered related.
+    min_samples : int, default=5
+        The number of samples required for a core point.
+    metric : string default='euclidean'
+        The used distance metric - more options are
+        'braycurtis', 'canberra', 'chebyshev', 'cityblock', 'correlation',
+        'cosine', 'dice', 'euclidean', 'hamming', 'jaccard', 'jensenshannon',
+        'kulsinski','manhattan'.
+    n_jobs : int or None, default=-1
+        The number of processes to start -1 means use all processors 
+    Attributes
+    ----------
+    labels : array, shape = [n_samples]
+        Cluster labels for the data - noise is defined as -1
+    """
```

Comment:
The docstring should reference the original implementation and/or the authors or papers, where relevant